### PR TITLE
Migrating from ftp to https

### PR DESCRIPTION
**Pull request by tapyu**
[`ftp` is deprecated by CDDIS since 2020](https://cddis.nasa.gov/Data_and_Derived_Products/CDDIS_Archive_Access.html). The current Matlab code was trying to download the RINEX file to extract the ephemeris data via `ftp` protocol. I changed it to `https`. I tried to use [`websave`](https://www.mathworks.com/help/matlab/ref/websave.html). However, I didn't work because now CDDIS requires you to login to the able to get the data. It seems that neither `websave` nor `curl` can handle the authentication step. `wget` made it, though. My unique concern is about the reproducibility of this code, I am not sure if Windows has `wget`. I am pretty sure that MacOS has it as it is a Unix-like system. I am on Linux and it worked seamlessly.

### CODE REVIEW

**Review by brianbreitsch — APPROVED** [review]
Thanks, I reviewed the changes and it looks good.  Indeed the old FTP access to CDDIS has been deprecated for quite some time.  I do not actively maintain this code anymore, so we appreciate your pull request.

**Comment by tapyu** [comment]
Great. Good to see that you quickly addressed to this PR. I've been working with [@rodrigodelimaf ](https://github.com/rodrigodelimaf) on obtaining the refractive and diffractive contribution of ionospheric scintillation separately. I suppose that, at this point, you already know him.

Since PR are welcoming I can try to create other PRs in order to update the code. I will only focus on organization and reproducibility.

File: Libraries/PropGeomCalc/ExtractRINEXeph.m [hunk 1]
```diff
@@ -16,10 +16,16 @@
...
+filename = ['brdc',DDD,'0.',YY,'n.Z'];
+RinexFile = [datadir,YYYY,'/',DDD,'/',YY,'n/',filename];
 [~,ephfile] = fileparts(RinexFile);
 if ~exist(ephfile,'file')
-    if ~exist([ephfile,'.Z'],'file') 
+    if ~exist([ephfile,'.Z'],'file')
+        username = input('Write the username: ', 's');
+        password = input('Write the password: ', 's');
+
+        system(['wget --auth-no-challenge --user=', username ' --password=', password, ' -O ', filename,' ', RinexFile])
```
Comment (tapyu) [inline Libraries/PropGeomCalc/ExtractRINEXeph.m:27]: For instance, I've tested on a MacOS and saw that, as Windows, it doesn't have `wget` (you can easily install it via `homebrew`, though, but it isn't built-in). Therefore, I will need to change this line to be able to download the RINEX file in other way. If I make it out, I will open other PRs :)

**Comment by brianbreitsch** [comment]
I'm not currently an active matlab user, so I'm not up-to-date on how to do downloads over HTTP, but a quick search shows you might be able to do something more cross-platform using `webread`.  I think authentication (username and password) can maybe be passed along to that function (in the options?).  Also, a better way to handle username/password going forward would be to check whether they are defined environment variables.  I recommend CDDIS_USERNAME and CDDIS_PASSWORD.

**Comment by tapyu** [comment]
The problem is that AFAIK `webread` is primarily for converting web content into a variable, and the URL used to download the RINEX (https://cddis.nasa.gov/archive/gnss/data/daily/2014/002/14n/brdc0020.14n.Z) is a compressed file. I tried to use, `websave`, which is designed for saving web content directly to a file. However, It seems not able to handle the CDDIS' authentication step.

**Comment by tapyu** [comment]
> Also, a better way to handle username/password going forward would be to check whether they are defined environment variables. I recommend CDDIS_USERNAME and CDDIS_PASSWORD.

Yes, much less intrusive. As soon as I find out how to download the file, I will go in this way.

**Pull request merged**